design/lifecycle.md

```diff
@@ -0,0 +1,195 @@
+
+### Spark Job proposal
+
+The user can run a spark job that is in charge of enforcing the lifecycle policy.
+
+The spark job will delete all the data (addresses) of objects matching the
+lifecycle rules except for data that is referenced by other objects (objects not
+matching the lifecycle rule). In order to find the matching objects efficiently
+(without going over the history of all commits every time) the spark job will
+conclude the Date lifecycle job ran last time. Using that Date, the job will
```

Comment:
Where does it store this data?  You know me, I would want it in a commit... but I assume others may want it as "business rules" metadata.  Either way, it has to go *somewhere*.